client/my-sites/plans-features-main/components/plan-upsell-modal/test/plan-upsell-modal.test.tsx

```diff
@@ -0,0 +1,129 @@
+/**
+ * @jest-environment jsdom
+ */
+import { PLAN_FREE, PLAN_PERSONAL } from '@automattic/calypso-products';
+import { screen } from '@testing-library/react';
+import useIsFreeDomainFreePlanUpsellEnabled from 'my-sites/plans-features-main/hooks/use-is-free-domain-free-plan-upsell-enabled';
+import useIsFreePlanCustomDomainUpsellEnabled from 'my-sites/plans-features-main/hooks/use-is-free-plan-custom-domain-upsell-enabled';
+import {
+	FREE_PLAN_FREE_DOMAIN_DIALOG,
+	FREE_PLAN_PAID_DOMAIN_DIALOG,
+	PAID_PLAN_IS_REQUIRED_DIALOG,
+} from '..';
+import { renderWithProvider } from '../../../../../test-helpers/testing-library';
+import { useModalResolutionCallback } from '../hooks/use-modal-resolution-callback';
+
+function MockPlansFeaturesMain( {
+	flowName,
+	selectedPlan,
+	paidSubdomain,
+}: {
+	flowName: string;
+	selectedPlan: string;
+	paidSubdomain: string;
+} ) {
+	const isCustomDomainAllowedOnFreePlan = useIsFreePlanCustomDomainUpsellEnabled(
+		flowName,
+		paidSubdomain
+	);
+	const isPlanUpsellEnabledOnFreeDomain = useIsFreeDomainFreePlanUpsellEnabled(
+		flowName,
+		paidSubdomain
+	);
+	const resolveModal = useModalResolutionCallback( {
+		isCustomDomainAllowedOnFreePlan,
+		isPlanUpsellEnabledOnFreeDomain,
+	} );
+	return <div data-testid="modal-render">{ resolveModal( selectedPlan ) }</div>;
+}
+
+describe( 'PlanUpsellModal tests', () => {
```

Comment:
@jeyip I went ahead and guarded those cases with unit tests. Now all we need is 1 manual test round 😄